lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -611,16 +611,19 @@ def add_auth_service(xml, payment_method, options)
 
       def add_normalized_threeds_2_data(xml, payment_method, options)
         threeds_2_options = options[:three_d_secure]
+        cc_brand = card_brand(payment_method).to_sym
 
-        xml.tag!('cavv', threeds_2_options[:cavv]) if threeds_2_options[:cavv] && card_brand(payment_method).to_sym != :master
+        xml.tag!('cavv', threeds_2_options[:cavv]) if threeds_2_options[:cavv] && cc_brand != :master
         xml.tag!('cavvAlgorithm', threeds_2_options[:cavv_algorithm]) if threeds_2_options[:cavv_algorithm]
         xml.tag!('paSpecificationVersion', threeds_2_options[:version]) if threeds_2_options[:version]
         xml.tag!('directoryServerTransactionID', threeds_2_options[:ds_transaction_id]) if threeds_2_options[:ds_transaction_id]
-        xml.tag!('commerceIndicator', options[:commerce_indicator] || ECI_BRAND_MAPPING[card_brand(payment_method).to_sym])
+        xml.tag!('commerceIndicator', options[:commerce_indicator] || ECI_BRAND_MAPPING[cc_brand])
         xml.tag!('eciRaw', threeds_2_options[:eci]) if threeds_2_options[:eci]
 
-        xid = threeds_2_options[:xid] || threeds_2_options[:cavv]
-        xml.tag!('xid', xid) if xid
+        if cc_brand == :visa
+          xid = threeds_2_options[:xid] || threeds_2_options[:cavv]
+          xml.tag!('xid', xid) if xid
```

Comment:
`xid` must be set regardless of the brand, even if it is not `visa`

Reply:
After chat IRL - I think we need to check what is the expectation fro 3DS 1.x

Reply:
Yup, while we wait for their response I added a commit to ensure that we send `xid` regardless of the card brand but allowing following Pierre's previous logic when it is visa and 3DS version 2.